base_layer/core/src/covenants/filters/fields_hashed_eq.rs

```diff
@@ -32,8 +31,9 @@ impl Filter for FieldsHashedEqFilter {
         let fields = context.next_arg()?.require_outputfields()?;
         let hash = context.next_arg()?.require_hash()?;
         output_set.retain(|output| {
-            let challenge = fields.construct_challenge_from(output);
-            Ok(challenge.finalize()[..] == *hash)
+            let challenge = fields.construct_challenge_from(output).finalize();
+            let challenge = challenge.to_vec();
+            Ok(challenge[..] == *hash)
```

Comment:
unnecessary allocation, this should work
```suggestion
            Ok(challenge.as_ref()[..] == *hash)
```

Reply:
Refactored, thanks for the suggestion !